### PR TITLE
remove abort when detect IPv6 support failed

### PR DESCRIPTION
**Pull request by virtualyw**
These's two reasons to avoid abort application since invalid socket check:
1. https://github.com/libuv/libuv/issues/1425

2. google search "create  socket  10106", found many similar issues that create socket failed

### CODE REVIEW

File: src/win/winsock.c [hunk 1]
```diff
@@ -131,21 +131,14 @@ void uv_winsock_init(void) {
...
-    /* Any error other than "socket type not supported" is fatal. */
-    uv_fatal_error(WSAGetLastError(), "socket");
+        SOL_SOCKET,
+        SO_PROTOCOL_INFOW,
+        (char*)&protocol_info,
+        &opt_len) == 0) {
+        if (!(protocol_info.dwServiceFlags1 & XP1_IFS_HANDLES))
+            uv_tcp_non_ifs_lsp_ipv6 = 1;
+    }
+    closesocket(dummy);
```
Comment (bnoordhuis) [inline src/win/winsock.c]: Shouldn't the closesocket() call still check for SOCKET_ERROR?

File: src/win/winsock.c [hunk 2]
```diff
@@ -131,21 +131,14 @@ void uv_winsock_init(void) {
...
-
-  } else if (!error_means_no_support(WSAGetLastError())) {
-    /* Any error other than "socket type not supported" is fatal. */
-    uv_fatal_error(WSAGetLastError(), "socket");
+        SOL_SOCKET,
+        SO_PROTOCOL_INFOW,
+        (char*)&protocol_info,
+        &opt_len) == 0) {
+        if (!(protocol_info.dwServiceFlags1 & XP1_IFS_HANDLES))
+            uv_tcp_non_ifs_lsp_ipv6 = 1;
```
Comment (bnoordhuis) [inline src/win/winsock.c]: Indent errors; please use two spaces.

File: src/win/winsock.c (hunk 1 above)
Comment (virtualyw) [reply src/win/winsock.c]: I think the return value is useless for a client application, we could do nothing when OS ipv6 configuration is not set correctly sometimes. Maybe the most import thing is Don't let it Crash :)

Comment (bnoordhuis) [reply src/win/winsock.c]: That's not the libuv philosophy. Unexpected errors shouldn't go unnoticed.

Comment (virtualyw) [reply src/win/winsock.c]: IPv6 socket is optional, we can use IPv4 instead,  so check of invalid IPv6 socket shouldn't be a fatal error.
It's not Unexpected errors.

**Comment by bzoz** [comment]
I'm generally +1 on this. I think it is better for libuv to say "no IPv6" than to die.

@virtualyw do you have access to a system where this error occurs? It looks like that requires some wild configuration. Could you confirm that libuv at least behaves somewhat reasonable in such cases? 

Also, could this be made more flat? There are 3 nested if-s, those can be made into single one, or maybe some gotos or maybe something like
```
uv_tcp_non_ifs_lsp_ipv6  = dummy != INVALID_SOCKET && getsockopt(...) && protocol_info...
```

**Comment by virtualyw** [comment]
@bzoz we have few chance to touch or remote control custom's computer,
maybe reset winsock will resolve the problem and we'll continue follow up it.
And I'm agree that the flat if-s should be better, but it's hard to found out a best form,should we talk in other topic next time?

File: src/win/winsock.c [hunk 3]
```diff
@@ -130,22 +130,15 @@ void uv_winsock_init(void) {
...
-  } else if (!error_means_no_support(WSAGetLastError())) {
-    /* Any error other than "socket type not supported" is fatal. */
-    uv_fatal_error(WSAGetLastError(), "socket");
+    errorno = getsockopt(dummy,
+        SOL_SOCKET,
+        SO_PROTOCOL_INFOW,
+        (char*)&protocol_info,
+        &opt_len);
+    if (errorno == 0 && !(protocol_info.dwServiceFlags1 & XP1_IFS_HANDLES)) {
+          uv_tcp_non_ifs_lsp_ipv6 = 1;
```
Comment (vtjnash) [inline src/win/winsock.c]: two space indent

File: src/win/winsock.c [hunk 4]
```diff
@@ -130,22 +130,15 @@ void uv_winsock_init(void) {
...
-      uv_tcp_non_ifs_lsp_ipv6 = 1;
-
-    if (closesocket(dummy) == SOCKET_ERROR)
-      uv_fatal_error(WSAGetLastError(), "closesocket");
-
-  } else if (!error_means_no_support(WSAGetLastError())) {
-    /* Any error other than "socket type not supported" is fatal. */
-    uv_fatal_error(WSAGetLastError(), "socket");
+    errorno = getsockopt(dummy,
+        SOL_SOCKET,
```
Comment (vtjnash) [inline src/win/winsock.c]: keep arguments aligned to `(`

**Comment by bnoordhuis** [comment]
@vtjnash Since you signed off on this: can you land this if you still think it LGTY?

**Comment by vtjnash** [comment]
Yes, though upon looking back, I decided to make a new PR to be even more lenient of errors here. I'm hoping to spend several days this week tackling libuv issues, limited only by getting reviewer time 🙂

**Comment by vtjnash** [comment]
Replaced by #2600

**Pull request closed without merging**